libcodechecker/server/api/product_server.py

```diff
@@ -128,48 +179,16 @@ def getProducts(self, product_endpoint_filter, product_name_filter):
                     util.escape_like(product_name_filter)), escape='*'))
 
             prods = prods.all()
-
             for prod in prods:
-                server_product = self.__server.get_product(prod.endpoint)
-
-                if not server_product:
-                    # TODO: Better support this, if the configuration database
-                    # is mounted to multiple servers?
-                    LOG.error("Product '{0}' was found in the configuration "
-                              "database, but no database connection is "
-                              "present. Was the configuration database "
-                              "connected to multiple servers?"
-                              .format(prod.endpoint))
-                    LOG.info("Please restart the server to make this "
-                             "product available.")
-                    continue
-
-                LOG.debug("Checking database connection for product: " +
-                          prod.endpoint)
-
-                server_product.connect()
-                db_status = server_product.db_status
-
-                name = base64.b64encode(prod.display_name.encode('utf-8'))
-                descr = base64.b64encode(prod.description.encode('utf-8')) \
-                    if prod.description else None
+                server_product, ret = self.__get_product(session, prod)
+                # Clients are expected to use this method to query if
+                # the product exists and usable. Usability sometimes requires
+                # "updating" the "is connected" status of the database.
+                if not server_product.connected:
```

Comment:
We should not depend on the connected value. It will be always False.

Reply:
This whole block is unneccessary since your migration script was added.